CourseApp/ClassTaskFolder/Document.cs

```diff
@@ -0,0 +1,40 @@
+namespace CourseApp
+{
+    using System;
+
+    public abstract class Document
+    {
+        private string _name;
+
+        public Document(string name)
+        {
+            Name = name;
+        }
+
+        public Document()
+        {
+            var generator = new RandomValueGenerator();
+            Name = generator.RandomName();
```

Comment:
Bad idea to use random generator in Constructor - you would not be able to control and test it